### ISSUE TITLE
Near 0 mAP from training

### ISSUE THREAD

**Issue by Hansmu** [google/automl#310]
Has anyone got a reasonable mAP after training a model? I've tried fiddling around with the parameters and used d0, d4, d6, yet the mAP from eval is barely above 0.

**Comment by elesun2018** [comment]
when i run eval code, meet a problem as follows:
automl0401/efficientdet# python main.py --mode=eval --model_name=${MODEL} --model_dir=${CKPT_PATH} --validation_file_pattern=tfrecord/val* --val_json_file=annotations/instances_val2017.json --hparams="use_bfloat16=false" --use_tpu=False

CKPT_PATH is the pretrained d0 model.


DONE (t=34.86s).
Average Precision (AP) @[ IoU=0.50:0.95 | area= all | maxDets=100 ] = 0.002
Average Precision (AP) @[ IoU=0.50 | area= all | maxDets=100 ] = 0.005
Average Precision (AP) @[ IoU=0.75 | area= all | maxDets=100 ] = 0.002
Average Precision (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.003
Average Precision (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.001
Average Precision (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets= 1 ] = 0.006
Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets= 10 ] = 0.011
Average Recall (AR) @[ IoU=0.50:0.95 | area= all | maxDets=100 ] = 0.012
Average Recall (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.018
Average Recall (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.006
Average Recall (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.004
INFO:tensorflow:Finished evaluation at 2020-04-28-17:12:04
INFO:tensorflow:Finished evaluation at 2020-04-28-17:12:04
I0428 17:12:04.599111 140494972888896 evaluation.py:275] Finished evaluation at 2020-04-28-17:12:04
INFO:tensorflow:Saving dict for global step 0: AP = 0.0022378154, AP50 = 0.0052162167, AP75 = 0.0015809378, APl = 0.00015473206, APm = 0.0014778649, APs = 0.0029822278, ARl = 0.0036031725, ARm = 0.006258475, ARmax1 = 0.005900326, ARmax10 = 0.011014862, ARmax100 = 0.011780873, ARs = 0.018089531, box_loss = 0.0, cls_loss = 0.02997985, global_step = 0, loss = 0.11981841
I0428 17:12:04.600037 140494972888896 estimator.py:2049] Saving dict for global step 0: AP = 0.0022378154, AP50 = 0.0052162167, AP75 = 0.0015809378, APl = 0.00015473206, APm = 0.0014778649, APs = 0.0029822278, ARl = 0.0036031725, ARm = 0.006258475, ARmax1 = 0.005900326, ARmax10 = 0.011014862, ARmax100 = 0.011780873, ARs = 0.018089531, box_loss = 0.0, cls_loss = 0.02997985, global_step = 0, loss = 0.11981841
INFO:tensorflow:Saving 'checkpoint_path' summary for global step 0: models/pretrained_models/efficientdet-d1/model

such low AP values ? why ?
thanks.

**Comment by Hansmu** [comment]
I'll close the ticket, as I'm getting acceptable mAP now. Pulled in the newest state and fixed my data some.

**Comment by elv-xuwen** [comment]
Hi @Hansmu , I'm trying to train D2 and D4 on TITAN RTX but the training process is slow because I can only set a small batch_size. Did you train on multi gpu or single gpu? Can you share your hyperparameters? Thank you so much!
